fiftyone/plugins/secrets.py

```diff
@@ -128,32 +146,36 @@ def __getitem__(self, key):
                 self.__secrets[key] = val
         return val
 
+    def __setattr__(self, key, value):
+        if re.search("_SecretsDictionary__(secrets|required_keys)$", key):
```

Comment:
Still don't see a point in preventing `__setattr__` here when the user can still do other things like `d._SecretsDictionary__secrets["bad"] = "thing"`.

Reply:
currently, there will be no error if you manually set the secret, however, it will not show up when you iterate through ctx.secrets.items()

Reply:
Would overriding `__getattr__` be the solution to reading/mutating the underlying dictionary?